code/game/gamemodes/blob/blobs/core.dm

```diff
@@ -1,93 +1,116 @@
+	return
+
+/obj/effect/blob/core/update_icon()
+	if(health <= 0)
+		qdel(src)
+		return
+	// update_icon is called when health changes so... call update_health in the overmind
+	if(overmind)
+		overmind.update_health()
+	return
+
+/obj/effect/blob/core/RegenHealth()
+	return // Don't regen, we handle it in Life()
+
+/obj/effect/blob/core/Life()
+	if(!overmind)
+		create_overmind()
+	else
+		if(resource_delay <= world.time)
+			resource_delay = world.time + 10 // 1 second
+			overmind.add_points(point_rate)
+	health = min(initial(health), health + 1)
+	if(overmind)
+		overmind.update_health()
+	for(var/i = 1; i < 8; i += i)
+		Pulse(0, i, overmind.blob_reagent_datum.color)
+	for(var/b_dir in alldirs)
+		if(!prob(5))
+			continue
+		var/obj/effect/blob/normal/B = locate() in get_step(src, b_dir)
+		if(B)
+			B.change_to(/obj/effect/blob/shield)
+			B.color = overmind.blob_reagent_datum.color
+	color = null
+	..()
+
+
+/obj/effect/blob/core/proc/create_overmind(var/client/new_overmind, var/override_delay)
+
+	if(overmind_get_delay > world.time && !override_delay)
 		return
 
-	Life()
-		if(!overmind)
-			create_overmind()
-		else
-			if(resource_delay <= world.time)
-				resource_delay = world.time + 10 // 1 second
-				overmind.add_points(point_rate)
-		health = min(initial(health), health + 1)
-		for(var/i = 1; i < 8; i += i)
-			Pulse(0, i)
-		for(var/b_dir in alldirs)
-			if(!prob(5))
-				continue
-			var/obj/effect/blob/normal/B = locate() in get_step(src, b_dir)
-			if(B)
-				B.change_to(/obj/effect/blob/shield)
-		..()
-
-
-	run_action()
-		return 0
-
-
-	proc/create_overmind(var/client/new_overmind)
-
-		if(overmind_get_delay > world.time)
-			return
-
-		overmind_get_delay = world.time + 300 // 30 seconds
-
-		if(overmind)
-			qdel(overmind)
-
-		var/client/C = null
-		var/list/candidates = list()
-
-		if(!new_overmind)
-			candidates = get_candidates(BE_BLOB)
-			if(candidates.len)
-				C = pick(candidates)
-		else
-			C = new_overmind
-
-		if(C)
-			respawnable_list -= C
-			var/mob/camera/blob/B = new(src.loc)
-			B.key = C.key
-			B.blob_core = src
-			src.overmind = B
-			if(!B.blob_core.creator)
-				B.verbs += /mob/camera/blob/proc/create_core
-			if(istype(ticker.mode, /datum/game_mode/blob))
-				var/datum/game_mode/blob/mode = ticker.mode
-				mode.infected_crew += B.mind
-			return 1
-		return 0
+	overmind_get_delay = world.time + 300 // 30 seconds
+
+	if(overmind)
+		qdel(overmind)
+
+	var/client/C = null
+	var/list/candidates = list()
+
+	if(!new_overmind)
+		candidates = get_candidates(BE_BLOB)
+		if(candidates.len)
+			C = pick(candidates)
+	else
+		C = new_overmind
+
+	if(C)
+		var/mob/camera/blob/B = new(src.loc)
+		B.key = C.key
+		B.blob_core = src
+		src.overmind = B
+		color = overmind.blob_reagent_datum.color
+		if(B.mind && !B.mind.special_role)
+			B.mind.special_role = "Blob Overmind"
+		spawn(0)
+			if(is_offspring)
+				B.verbs -= /mob/camera/blob/verb/split_consciousness

```

Comment:
Noope, trust me, it needs this.

Tg blobs used to have offspring splitting, it ended with RAPID MITOSIS BLOB.

Balance shows this is needed.